Subsurface/Node/Common/Peripherals/inc/PWMManager.h

```diff
@@ -0,0 +1,57 @@
+#include "PWMInterface.h"
+#include <cstdint>
+//#include "mock_PWMInterface.h"
+
+#ifndef PWMMANAGER_H
+#define PWMMANAGER_H
+
+class PWMManager
+{
+public:
+    
+    // Constructor
+    PWMManager(uint64_t clockSpeed, uint32_t preScaler, uint32_t period, 
+    uint32_t pulse, PWMInterface interface, TIM_HandleTypeDef timer, uint32_t channel);
```

Comment:
So I think we will only need to take a `PWMInterface*` here. The idea is that the interface class handles all calls to HAL functions and this manager class is used to interact with the interface class.